hkdf.go

```diff
@@ -208,6 +208,11 @@ func ExpandHKDFOneShot(h func() hash.Hash, pseudorandomKey, info []byte, keyLeng
 			return nil, err
 		}
 		defer ossl.EVP_PKEY_CTX_free(ctx)
+		if len(out) == 0 {
+			// Nothing to do, and some engines may error on zero-length output.
+			// Call newHKDFCtx1, tought, to validate parameters.
```

Comment:
Typo `tought`, but I also think the reasoning can be a bit clearer:

```suggestion
			// Nothing to do, so exit early.
			// We also can't call EVP_PKEY_derive because some engines error on zero-length output.
			// We can only exit after calling newHKDFCtx1 because we still need it to validate the parameters.
```